src/platform/plugins/shared/workflows_management/public/widgets/workflow_yaml_editor/lib/snippets/insert_trigger_snippet.test.ts

```diff
@@ -104,4 +104,462 @@ describe('insertTriggerSnippet', () => {
+            range: new monaco.Range(2, 1, 3, 1),
+            text: `  ${expectedSnippet}`,
+          },
+        ],
+        expect.any(Function)
+      );
+    });
+
+    it('should replace empty item with comment', () => {
+      const inputYaml = `triggers:\n  - # comment\n`;
+      const model = createFakeMonacoModel(inputYaml);
+      const yamlDocument = parseDocument(inputYaml);
+
+      insertTriggerSnippet(model as unknown as monaco.editor.ITextModel, yamlDocument, 'manual');
+
+      expect(generateTriggerSnippetSpy).toHaveBeenCalledWith('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+        withTriggersSection: false,
+      });
+
+      const expectedSnippet = generateTriggerSnippetModule.generateTriggerSnippet('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+      });
+
+      expect(model.pushEditOperations).toHaveBeenCalledWith(
+        null,
+        [
+          {
+            range: new monaco.Range(2, 1, 3, 1),
+            text: `  ${expectedSnippet}`,
+          },
+        ],
+        expect.any(Function)
+      );
+    });
+
+    it('should replace empty item with trailing spaces and comment', () => {
+      const inputYaml = `triggers:\n  -  # comment\n`;
+      const model = createFakeMonacoModel(inputYaml);
+      const yamlDocument = parseDocument(inputYaml);
+
+      insertTriggerSnippet(model as unknown as monaco.editor.ITextModel, yamlDocument, 'manual');
+
+      expect(generateTriggerSnippetSpy).toHaveBeenCalledWith('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+        withTriggersSection: false,
+      });
+
+      const expectedSnippet = generateTriggerSnippetModule.generateTriggerSnippet('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+      });
+
+      expect(model.pushEditOperations).toHaveBeenCalledWith(
+        null,
+        [
+          {
+            range: new monaco.Range(2, 1, 3, 1),
+            text: `  ${expectedSnippet}`,
+          },
+        ],
+        expect.any(Function)
+      );
+    });
+  });
+
+  describe('when triggers section exists but is empty', () => {
+    it('should insert trigger when triggers: [] (empty array)', () => {
+      const inputYaml = `triggers: []`;
+      const model = createFakeMonacoModel(inputYaml);
+      const yamlDocument = parseDocument(inputYaml);
+
+      insertTriggerSnippet(model as unknown as monaco.editor.ITextModel, yamlDocument, 'manual');
+
+      expect(generateTriggerSnippetSpy).toHaveBeenCalledWith('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+        withTriggersSection: false,
+      });
+
+      const expectedSnippet = generateTriggerSnippetModule.generateTriggerSnippet('manual', {
+        full: true,
+        monacoSuggestionFormat: false,
+      });
+
+      expect(model.pushEditOperations).toHaveBeenCalledWith(
+        null,
+        [
+          {
+            range: new monaco.Range(2, 1, 2, 1),
```

Comment:
when inserting new trigger to `triggers: []`, we should remove `[]`


https://github.com/user-attachments/assets/e90372ae-b182-49e4-8038-7438e8162c03